### ISSUE TITLE
Typescript definitions

### ISSUE THREAD

**Issue by nitech** [angular-ui/ng-grid#3720]
I can't seem to find any TypeScript definitions for ui-grid. Is there one?

**Comment by JLLeitschuh** [comment]
No. ui grid is written in Angular 1 not Angular 2. Angular 1 is entirely html/css/javascript.

**Comment by nitech** [comment]
Yes, I realize that. A lot of people use TypeScript with Angular 1. A TypeScript definitions file for ui-grid would strengthen its position in the market.

**Comment by JLLeitschuh** [comment]
If you know how to add it a pull request would be appreciated.

**Comment by nitech** [comment]
If we end up using ui-grid, I will start writing one. Still evaluating
different grids :-)

mvh
Simon Pedersen
466 34 123

**Comment by DmitryEfimenko** [comment]
+1.
I realize there is a lot of code written already for ui-grid, but writing code in TypeScript has a lot of benefits (as angular core team suggests) and it really has nothing to do with the fact that ui-grid is written for angular 1, not 2.
I'd recommend looking into TS in general if you are not familiar with it yet.

If someone started working on d.ts file, please share the work.

**Comment by JLLeitschuh** [comment]
How much work is there to implement a compiler as a build step?

**Comment by DmitryEfimenko** [comment]
adding a compilation step to the build process usually is a trivial task.
I created a [d.ts file here](https://github.com/DmitryEfimenko/angular-ui-grid.d.ts/blob/master/angular-ui-grid.d.ts), which covers most methods, events, and properties for all features.

It'd be great if a UiGrid developer could review this before I submit it to [DefinitelyTyped](https://github.com/borisyankov/DefinitelyTyped).

**Comment by JLLeitschuh** [comment]
I'm not willing to add a statically created typescript file. Is there any way that we could make this dynamic. Our API is subject to change and trying to keep a typed file updated is thwart with unhappy issue openers.

**Comment by DmitryEfimenko** [comment]
Of course there is a way. I'm sure Angular team is doing it somehow.
Some information to consider:
* [JavaScript doesn't always contain enough type information for the TypeScript compiler](http://stackoverflow.com/a/18302200/894273)
* [Aurelia has a design doc for doing this](https://github.com/cmichaelgraham/aurelia-typescript/blob/master/generating-aurelia-type-definitions.md)

From what I can see - the .js files would have to be renamed to .ts and we'd go from there.

**Comment by JLLeitschuh** [comment]
This is not something the core team has the time to dedicate to adding.
Enhancements are usually added because someone takes the time to add it and make a pull request.
Come chat with us in gitter.
[![Gitter](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/angular-ui/ui-grid?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)

**Comment by mkalinov** [comment]
Hi guys, 
Is it what u'r looking for?
https://www.npmjs.com/package/@types/ui-grid    => https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/ui-grid
